source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -6,38 +6,27 @@
 #include <MaterialXGenShader/TypeDesc.h>
 
 #include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/GenContext.h>
 
 MATERIALX_NAMESPACE_BEGIN
 
 namespace
 {
 
-using TypeDescMap = std::unordered_map<string, TypeDesc>;
 using TypeDescNameMap = std::unordered_map<uint32_t, string>;
 
-// Internal storage of registered type descriptors
-TypeDescMap& typeMap()
-{
-    static TypeDescMap map;
-    return map;
-}
-
+// We still need to keep a static map of the type names, because we don't want to store the
+// names inside the TypeDesc object, but we need to be able to implement TypeDesc::getName(),
+// without access to the GenContext object, and thus no access to TypeDescStorage.
+// We could perhaps consider using OIIO::ustring instead of a hash for TypeDesc::_id, this
+// wouldn't remove the static, but move it to OIIO, but the OIIO ustring implementation is
+// really robust and battle tested.
```

Comment:
This makes sense. The reason it's implemented this way now (using a `std::string_view` for the initialization/registration, which gets stored as a hash and a static name map) is because of the lack of a good class for string interning. If we can allow a dependency on OIIO, using `ustring` instead would be a nice improvement.

Reply:
I actually hit CI issues with static lib builds and the typenamemap being static - so I've got an upcoming patch that moves the typenamemap in to the TypeDescStorage as well. 

I still think it's an interesting idea to start using OIIO::ustrings - it would be nice if there was a header only version of this we could adopt and not need full library dependency.

Reply:
Expressing it in a header that has no other dependencies -- that could maybe be arranged? But there's still a singleton at the heart of it, it can't be expressed as completely inline.